resources/mysql.go

```diff
@@ -0,0 +1,240 @@
+package resources
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"database/sql"
+	"embed"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	// install mysql driver
+	_ "github.com/go-sql-driver/mysql"
+)
+
+// MysqlInstance represents a mysql instance.
+type MysqlInstance struct {
+	dir  string
+	port int
+	proc *os.Process
+}
+
+// CreateMysqlInstance creates a mysql instance.
+// Must call Purge() to remove the mysql instance after use.
+func CreateMysqlInstance() (*MysqlInstance, error) {
+	basedir, err := installMysql8()
+	if err != nil {
+		return nil, err
+	}
+	if err := initMysql8(basedir); err != nil {
+		return nil, err
+	}
+	return startMysql8(basedir)
+}
+
+// Port returns the port of the mysql instance.
+func (instance MysqlInstance) Port() int { return instance.port }
+
+// Purge removes the mysql instance.
+func (instance *MysqlInstance) Purge() error {
+	if err := instance.proc.Kill(); err != nil {
+		return err
+	}
+	return os.RemoveAll(instance.dir)
+}
+
+//go:embed mysql-8.0.28-macos11-arm64.tar.gz mysql-8.0.28-linux-glibc2.17-x86_64-minimal.tar.xz
```

Comment:
This might put both pg and MySQL into Bytebase binary since they are in the same package. Shall we create a new package? Also can we reuse some functions if possible?

Reply:
done